app/scripts/directives/createProject.js

```diff
@@ -0,0 +1,53 @@
+"use strict";
+
+angular.module("openshiftConsole")
+
+  .directive("createProject", function() {
+    return {
+      restrict: 'E',
+      scope: {
+        alerts: '=',
+        submitButtonLabel: '@',
+        redirectAction: '&'
+      },
+      templateUrl: 'views/directives/_create-project-form.html',
+      controller: function($scope, $filter, $location, DataService) {
+        if(!($scope.submitButtonLabel)) {
+          $scope.submitButtonLabel = 'Create';
+        }
+        $scope.createProject = function() {
```

Comment:
We typically make the directive just the editor that takes in the object and updates it, then let the controller call create and handle errors.